lib/ansible/inventory/data.py

```diff
@@ -158,6 +158,7 @@ def get_host(self, hostname):
     def add_group(self, group):
         ''' adds a group to inventory if not there already '''
 
+        group = to_safe_group_name(group)
```

Comment:
Yeah, AlanCoding makes a good point. It seems like the conversion should stay where it was for the time being, with a deprecation warning in `Group.__init__`. You could fix the composed groups with hyphens bugs before the add_group there. I like the finer grain of control this gives rather than a toggle.

Reply:
i expected my change to show deficiencies in the existing plugins, we need to fix them instead of 'passing on' the responsibility to the plugin author or we would allow invalid names into the system that can blow up randomly later. It is better to break 'at origin'.

Reply:
this was my mistake in the initial design as the 'core/common' code should always have been the point of enforcement.

Reply:
We don't have tests for inventory scripts, so it's not clear how many things this breaks.

Reply:
i pushed 'fix' for script plugin, i can add tests (thought we had simple one)

Reply:
I meant that we don't have tests for specific scripts that may or may not be generating inventory with '-' in group names. There are a couple correctly sanitizing group names, a couple doing it inadequately, and a number that appear to never sanitize. Adding the deprecation period resolves my concern since now the scripts have forewarning before they break.

Reply:
i'm mostly exploring what the fix should look like, i'm not sure the current way is best since it doesn't cover 3rd party inventory plugins (but i expect almost none exist currently)